test/widgets/search_ride_page_test.dart

```diff
@@ -0,0 +1,972 @@
+              .map((Feature feature) => ProfileFeatureFactory().generateFake(feature: feature))
+              .toList(),
+        );
+        final Profile noRatings = ProfileFactory().generateFake(
+          reviewsReceived: [],
+          profileFeatures: requiredFeatures
+              .map((Feature feature) => ProfileFeatureFactory().generateFake(feature: feature))
+              .toList(),
+        );
+        final Profile notEnoughFeatures = ProfileFactory().generateFake(
+          reviewsReceived: [
+            ReviewFactory().generateFake(
+              rating: minRating,
+              comfortRating: NullableParameter(minRating),
+              safetyRating: NullableParameter(minRating),
+              reliabilityRating: NullableParameter(minRating),
+              hospitalityRating: NullableParameter(minRating),
+            ),
+          ],
+          profileFeatures: requiredFeatures
+              .map((Feature feature) => ProfileFeatureFactory().generateFake(feature: feature))
+              .toList()
+              .sublist(1),
+        );
+
+        final List<Profile> drivers = [
+          satisfiesEverything,
+          notEnoughCategoryRating,
+          notEnoughOverallRating,
+          noRatings,
+          notEnoughFeatures,
+        ];
+
+        //I am setting the duration and positions here because default Dart sort isn't stable and I want indices to work
+        final List<Drive> drives = drivers
+            .map((Profile driver) => DriveFactory().generateFake(
+                startTime: startTime,
+                endTime: startTime.add(Duration(minutes: drivers.indexOf(driver) + 1)),
+                startPosition: Position(0, 0),
+                endPosition: Position(0, 0),
+                driver: NullableParameter(driver)))
+            .toList();
+        final List<Map<String, dynamic>> driveJsons = drives.map((Drive drive) => drive.toJsonForApi()).toList();
+
+        whenRequest(processor).thenReturnJson(driveJsons);
+        for (final Drive drive in drives) {
+          whenRequest(processor, urlMatcher: matches(RegExp('/rest/v1/drives.*id=eq.${drive.id}')))
+              .thenReturnJson(drive.toJsonForApi());
+        }
+
+        await pumpMaterial(tester, const SearchRidePage());
+
+        final SearchRidePageState pageState = tester.state(pageFinder);
+
+        await enterStartAndDestination(tester, faker.address.city(), faker.address.city());
+
+        //Only ratings, shows 3 drives: Satisfies everything, no ratings, not enough features
+        await enterFilter(
+          tester,
+          rating: minRating,
+          comfortRating: minRating,
+          safetyRating: minRating,
+          reliabilityRating: minRating,
+          hospitalityRating: minRating,
+        );
+
+        List<Ride> filteredRides = pageState.filter.apply(pageState.rideSuggestions!, pageState.selectedDate);
+        expect(filteredRides, hasLength(3));
+        expect(filteredRides[0].driveId, drives[0].id);
```

Comment:
Still not happy you need to use indices here instead of the name, but I guess there is no way to get to the drive of that driver?

Reply:
Maybe use `Map.fromIterables`?